### PR TITLE
GM service support as resource

### PR DESCRIPTION
**Pull request by asarfaty**

### CODE REVIEW

File: nsxt/resource_nsxt_policy_service.go [hunk 1]
```diff
@@ -433,9 +440,28 @@ func resourceNsxtPolicyServiceCreate(d *schema.ResourceData, m interface{}) erro
...
 	log.Printf("[INFO] Creating service with ID %s", id)
-	err = client.Patch(id, obj)
-	if err != nil {
-		return handleCreateError("Service", id, err)
+
+	if isPolicyGlobalManager(m) {
+		converter := bindings.NewTypeConverter()
+		converter.SetMode(bindings.REST)
+
+		dataValue, err1 := converter.ConvertToVapi(obj, model.ServiceBindingType())
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_service.go:448]: I've made a utility for this - I can replace this later.

File: nsxt/resource_nsxt_policy_service.go [hunk 2]
```diff
@@ -433,9 +440,28 @@ func resourceNsxtPolicyServiceCreate(d *schema.ResourceData, m interface{}) erro
...
+		}
+		gmService := gmObj.(gm_model.Service)
+		client := gm_infra.NewDefaultServicesClient(connector)
+		err = client.Patch(id, gmService)
+	} else {
+		client := infra.NewDefaultServicesClient(connector)
+		err = client.Patch(id, obj)
+		if err != nil {
+			return handleCreateError("Service", id, err)
+		}
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_service.go:464]: looks like this should move out of the global manager if?

File: nsxt/resource_nsxt_policy_service.go [hunk 3]
```diff
@@ -433,9 +440,28 @@ func resourceNsxtPolicyServiceCreate(d *schema.ResourceData, m interface{}) erro
...
+		if err2 != nil {
+			return err2[0]
+		}
+		gmService := gmObj.(gm_model.Service)
+		client := gm_infra.NewDefaultServicesClient(connector)
+		err = client.Patch(id, gmService)
+	} else {
+		client := infra.NewDefaultServicesClient(connector)
+		err = client.Patch(id, obj)
+		if err != nil {
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_service.go:462]: looks like this should move out of the global manager if?

File: nsxt/resource_nsxt_policy_service.go (hunk 1 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_policy_service.go:448]: Sure. Thanks.

File: nsxt/resource_nsxt_policy_service.go (hunk 3 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_policy_service.go:462]: Fixed

**Pull request merged**